src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -595,53 +596,67 @@ internal bool IsEntryPointCandidate
             get { return IsStatic && Name == WellKnownMemberNames.EntryPointMethodName; }
         }
 
+
+        internal bool ReturnsAwaitableToVoidOrInt(CSharpCompilation compilation, DiagnosticBag diagnostics)
```

Comment:
Consider moving `ReturnsAwaitableToVoidOrInt` and `HasEntryPointSignature` to `CSharpCompilation` as `private` methods.